gradio/events.py

```diff
@@ -78,703 +96,123 @@ def change(
             warnings.warn(
                 "The 'status_tracker' parameter has been deprecated and has no effect."
             )
-        dep = self.set_event_trigger(
-            "change",
+        dep, dep_index = self.trigger.set_event_trigger(
+            self.event_name,
             fn,
             inputs,
             outputs,
             preprocess=preprocess,
             postprocess=postprocess,
             scroll_to_output=scroll_to_output,
-            show_progress=show_progress,
+            show_progress=show_progress
+            if show_progress is not None
+            else self.show_progress,
             api_name=api_name,
             js=_js,
             queue=queue,
             batch=batch,
             max_batch_size=max_batch_size,
             every=every,
+            after=self.after,
         )
-        set_cancel_events(self, "change", cancels)
-        return dep
+        set_cancel_events(self.trigger, self.event_name, cancels)
+        return Dependency(self.trigger, dep_index, dep)
+
+
+class Changeable(EventListener):
+    def __init__(self):
+        self.change = EventListenerMethod(self, "change")
+        """
+        This event is triggered when the component's input value changes (e.g. when the user types in a textbox
+        or uploads an image). This method can be used when this component is in a Gradio Blocks.
+        """
 
 
 class Clickable(EventListener):
-    def click(
-        self,
-        fn: Callable | None,
-        inputs: Component | List[Component] | Set[Component] | None = None,
-        outputs: Component | List[Component] | None = None,
-        api_name: str | None = None,
-        status_tracker: StatusTracker | None = None,
-        scroll_to_output: bool = False,
-        show_progress: bool = True,
-        queue=None,
-        batch: bool = False,
-        max_batch_size: int = 4,
-        preprocess: bool = True,
-        postprocess: bool = True,
-        cancels: Dict[str, Any] | List[Dict[str, Any]] | None = None,
-        every: float | None = None,
-        _js: str | None = None,
-    ):
+    def __init__(self):
```

Comment:
The typehints are preserved which is awesome but we're losing the docstring.

### Main
<img width="905" alt="image" src="https://user-images.githubusercontent.com/41651716/223586379-9ebd05ae-17ee-4ceb-ab9e-dfb920ff1292.png">


### This branch
<img width="921" alt="image" src="https://user-images.githubusercontent.com/41651716/223586264-b493e2bc-6213-4bed-ba7e-763516ab3caa.png">


I think a good workaround would be manually adding the docstring of `EventListenerMethod` to the docstring you already have here. For example,

```python

    def __init__(self):
        self.click = EventListenerMethod(self, "click")
        f"""
        This event is triggered when the component (e.g. a button) is clicked.
        This method can be used when this component is in a Gradio Blocks.
        
        {self.click.__doc__}
        """
```

Reply:
clever! will do

Reply:
hmm I still didn't get the type hint - does it work for you @freddyaboulton?

Reply:
btw the behaviour in this branch rn for me is: I get the docstring while typing `click`, and argument type hints when I open the parentheses.

Reply:
Dang yea thought this was working for me. Let me think about this some more, I think the problem is that the `click` is determined at runtime. Should be fine to proceed without this though?

Reply:
I get the same thing @aliabid94. Docstring while typing in `.click()`: 

<img width="770" alt="image" src="https://user-images.githubusercontent.com/1778297/223593578-2298234d-dc3d-4767-bf63-8892ca7a35b7.png">

Type hints after opening the parentheses:

<img width="498" alt="image" src="https://user-images.githubusercontent.com/1778297/223593629-d1f58a25-18f9-4dd5-ac69-52b173e9a0b8.png">

Very cool!